### DISCUSSION TITLE
Migrating from prometheus-net to OpenTelemetry .NET

### DISCUSSION THREAD

**Discussion by joacar** [open-telemetry/opentelemetry-dotnet#4248] [Q&A]
Hi

Currently I'm using the promtheus-net package which is added as a middleware and metrics are exported. The name of the metrics are
* http_requests_received_total
* http_requests_in_progress
* http_request_duration_seconds_{sum,count,bucket}

In OT the only one I can see is `http_server_duration_ms`. What do I need to do to make in progress and total available? Looking at the code it does seem to hook into the `HttpRequestIn` events in diagnostic listener. Perhaps I need to manually add the source `System.Net.Http`?
If there is any source mentioning how to migrate from prometheus-net to OT I would be very happy. If not, I might create one :)

I found [this](https://github.com/open-telemetry/opentelemetry-dotnet/discussions/4165) that is related but doesn't answer my question.

Thanks

**Comment by cijothomas** [comment]
_What do I need to do to make in progress and total available?_ 

The metrics from OTel follow OTel conventions.
1. Total request - is part of the http.server.duration (its a histogram, so you can get count)
2. In-progress is not yet implemented. Its part of the Otel metric conventions, but we just haven't implemented it.

The names/dimensions etc. will quite likely be different from Prometheus.NET. (I'd suggest to look into using Otel-Collector, which might have something to rename metrics etc, so that you get same names, if that is what you are looking for).

**Reply by joacar** [reply]
Perfect thank you! Yeah I haven't thought about naming, my concern now is to get the experimental app show same data in Grafana as the others. Then whether to change in OT or move to use new naming is a later question :)
Do you need someone to take a look at implementing it? In one of my applications I wrote my own prometheus-net based on `DiagnosticListener` instead of middleware, I guess this is the way it should be treated in OT as well.

Cheers

**Reply by cijothomas** [reply]
@vishweshbankwar can help give some pointers, if you are interested in helping us add the in-progress counter. (most of the infra is already there, but have to incorporate)
